### PR TITLE
Improve the integration test asserts

### LINKED ISSUE #20: Check integration tests response bodies.
Check response bodies and/or the database after an action

### PR DESCRIPTION
**Pull request by DeKleineKobini**
#20

### CODE REVIEW

File: inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/AuthenticationControllerIT.java [hunk 1]
```diff
@@ -64,7 +49,14 @@ void loginWithExistingCredentialsReturning200() throws Exception {
...
                 )
-                .andExpect(status().isOk());
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.accessToken").isNotEmpty())
+                .andExpect(jsonPath("$.tokenType").value("Bearer"))
+                .andExpect(jsonPath("$.id").value(account.getId()))
+                .andExpect(jsonPath("$.email").value("test@inetum-realdolmen.world"))
+                .andExpect(jsonPath("$.roles").isArray())
+                .andExpect(jsonPath("$.roles.length()").value(1))
+                .andExpect(jsonPath("$.roles[0]").value("ROLE_JOBSEEKER"));
```
Comment (TomD011099) [inline inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/AuthenticationControllerIT.java]: Maybe use andExpectAll?

File: inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/VacancyControllerIT.java [hunk 2]
```diff
@@ -19,27 +14,28 @@
...
+                                .queryParam("functionTitle", "")
+                                .queryParam("contractType", "")
+                                .queryParam("industry", "")
+                                .queryParam("country", "726")
+                                .queryParam("requiredYearsOfExperience", "0")
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isNoContent())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$").isEmpty());
```
Comment (TomD011099) [inline inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/VacancyControllerIT.java]: Maybe use andExpectAll?

File: inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/VacancyControllerIT.java [hunk 3]
```diff
@@ -62,16 +58,42 @@ public void findAllVacanciesWithFilter_GETBelgium3YearsOfExpVacancy_Success() th
...
+                                .queryParam("contractType", "")
+                                .queryParam("industry", "")
+                                .queryParam("country", String.valueOf(country1.getId()))
+                                .queryParam("requiredYearsOfExperience", "3")
                                 .contentType(MediaType.APPLICATION_JSON)
                 )
-                .andExpect(status().isOk());
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$.length()").value(3));
```
Comment (TomD011099) [inline inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/VacancyControllerIT.java]: Maybe use andExpectAll?

File: inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/VacancyControllerIT.java [hunk 4]
```diff
@@ -112,7 +136,9 @@ public void addVacancy_POSTValidVacancy_Success() throws Exception {
                                 .contentType(MediaType.APPLICATION_JSON)
                                 .content(asJsonString(vacancy))
                 )
-                .andExpect(status().isOk());
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").isNotEmpty())
+                .andExpect(jsonPath("$.functionTitle").value("Vendor"));
```
Comment (TomD011099) [inline inetum-realjobs-backend/src/test/java/world/inetum/realdolmen/realjobs/controllers/VacancyControllerIT.java]: Maybe use andExpectAll?

**Review by TomD011099 — CHANGES_REQUESTED** [review]

**Review by TomD011099 — APPROVED** [review]

**Review by redasaada — APPROVED** [review]

**Pull request merged**